### DISCUSSION TITLE
Posting text snippets instead of only messages

### DISCUSSION THREAD

**Discussion by shoqvalue** [element-hq/element-meta#692] [Ideas]
#### What would you like to do?
Be able to post collapsible snippets of text, as is common on Slack and some of its clones.

#### Why would you like to do it?
Because often the entire need is to share a chunk of text, a transcript, or code with the team, and being able to do that quickly and easily would not only be an important feature, but one that most Slack users have come to look for when they look for alternative platforms

#### How would you like to achieve it?
Pretty much the way slack does. Click  a "Plus+" sign to add content instead of a message.  Just as the Slack permits here.  I assume storing the snippet is not a big issue.   

Full-blown page "posts" would be nice too, but snippets are something that will do in the interim.

![2022-05-18_13h22_39](https://user-images.githubusercontent.com/10272750/169105655-bed49eeb-e92f-414f-8b2a-5265a4eabf05.png)

### Have you considered any alternatives?

I don't think there's any other approach that makes sense. It's a pretty basic feature I would use a dozen times a day (as I do on Slack now).

**Comment by robintown** [comment]
Alternatives include:
- Uploading the content as a file
- Pasting it into a code block, which makes it collapsible and easily copied

**Comment by shoqvalue** [comment]
IMO, that's just a workaround, not an alternative. The entire point to a snippet is to paste a chunk of text in a second or two while the need is top of mind, not having to take the time to create a file (I haven't had to do that in years to share a chunk of text). That's why snippets are so popular.   

As for "pasting it into a code block," that would be acceptable if a) it didn't automatically format plain text like code, and b) I had that option in my interface.  I don't (or I am missing it somewhere). Do you?

**Comment by robintown** [comment]
Thanks, that helps understand the request a bit better.

Highlighted text can be formatted as a code block like so:
![code](https://user-images.githubusercontent.com/48614497/169112154-ac26303b-79dc-4bb4-b1ef-1651f529efa8.png)

**Comment by shoqvalue** [comment]
Oh, highlighted text. Gotcha,. Yeah, that's useful, but a snippet is more useful for code (imo) because it can be edited and annotated like a document right there in the message editor.
\

**Comment by dkasak** [comment]
I often want to share my Markdown notes with people, but the note can get long and then it feels spammy when it fills out an entire screenful or more. This would come very handy in this case.

If text snippets are implemented, one other useful behaviour change for them would be to parse Markdown properly, so that paragraphs are delimited with empty lines between them and newline characters inside the paragraph are ignored. Currently Element inserts a `<br>` for each newline, reportedly because this is the default expectation for a chat client, but this is contrary to all Markdown standards (including Commonmark, which Element is supposed to be following).

**Comment by kittykat** [comment]
Hi, I'm going to move this issue to Element meta discussions as we need to consider it cross-platform :+1:

**Reply by shoqvalue** [reply]
Glad to hear this.

**Comment by Julian-Dumitrascu** [comment]
> Full-blown page "posts"
What are these? 
What would you like to do? How would you like the software to help?
